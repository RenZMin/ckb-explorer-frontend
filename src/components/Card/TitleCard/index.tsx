import { TitleCardPanel } from './styled'

export default ({ title, isSingle }: { title: string; isSingle?: boolean }) => (
  <TitleCardPanel isSingle={isSingle}>
    <div className="title__card__content">{title}</div>
  </TitleCardPanel>
)
