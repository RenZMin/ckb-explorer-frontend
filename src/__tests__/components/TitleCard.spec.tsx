import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import TitleCard from '../../components/Card/TitleCard'

describe('TitleCard Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <TitleCard title="title" />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find('.title__card__content')).toHaveLength(1)
    expect(wrapper.find('.title__card__content').text()).toBe('title')
  })
})
