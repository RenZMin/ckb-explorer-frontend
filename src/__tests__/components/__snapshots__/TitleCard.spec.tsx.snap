// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleCard Component shallow renders 1`] = `
<div
  className="sc-bdnxRM crdggW"
>
  <div
    className="title__card__content"
  >
    title
  </div>
</div>
`;
